components/fd/component_output.go

```diff
@@ -122,72 +161,97 @@ var (
 // only set once since it relies on the kube client and specific port
 func setDefaultPoller(cfg Config) {
 	defaultPollerOnce.Do(func() {
-		defaultPoller = query.New(Name, cfg.Query, Get)
+		defaultPoller = query.New(Name, cfg.Query, CreateGet(cfg))
 	})
 }
 
 func getDefaultPoller() query.Poller {
 	return defaultPoller
 }
 
-func Get(ctx context.Context) (_ any, e error) {
-	defer func() {
-		if e != nil {
-			components_metrics.SetGetFailed(Name)
-		} else {
-			components_metrics.SetGetSuccess(Name)
+func CreateGet(cfg Config) query.GetFunc {
```

Comment:
Is this possible to decouple calculation logics from low level syscalls or file access and write UTs?

Reply:
done